spec/es_spec_helper.rb

```diff
@@ -67,19 +67,24 @@ def self.es_version
   end
 
   RSpec::Matchers.define :have_hits do |expected|
+    hits_count_path = ESHelper.es_version_satisfies?(">=7") ? %w(hits total value) : %w(hits total)
+
     match do |actual|
-      if ESHelper.es_version_satisfies?(">=7")
-        expected == actual['hits']['total']['value']
-      else
-        expected == actual['hits']['total']
-      end
+      @actual_hits_count = actual&.dig(*hits_count_path)
+      values_match? expected, @actual_hits_count
+    end
+    failure_message do |actual|
```

Comment:
review note: the changes to the es_spec_helper make previously-flaky specs have more understandable causes, but those specs are no longer flaky :)